docs/NOW-spec.md

```diff
@@ -1597,8 +1265,52 @@ The NOW_EXEC_PWSH_MSG message is used to execute a remote PowerShell 7 (pwsh) co
 
 **sessionId (4 bytes)**: A 32-bit unsigned integer containing a unique remote execution session id.
 
-**executionPolicy (variable)**: A NOW_VARSTR structure, same as with NOW_EXEC_WINPS_MSG.
+**command (variable)**: A NOW_VARSTR structure containing the command to execute.
 
-**configurationName (variable)**: A NOW_VARSTR structure, same as with NOW_EXEC_WINPS_MSG.
+**executionPolicy (variable, optional)**: A NOW_VARSTR structure, same as with NOW_EXEC_WINPS_MSG.
 
-**command (variable)**: A NOW_VARSTR structure containing the command to execute.
+**configurationName (variable, optional)**: A NOW_VARSTR structure, same as with NOW_EXEC_WINPS_MSG.
+
+#### NOW_EXEC_HEARTBEAT_MSG
```

Comment:
Do we really need an heartbeat for that? Since this is a virtual channel via RDP transport, my understanding is that the (RDP) session is necessarily alive. Am I missing something?

Reply:
Hmm, I think you are right, a _per-session_ periodical heartbeat indeed is unnecessary. My reasoning for adding it was to detect when the specific session processing has failed unexpectedly without sending ExecResult, to avoid unresponsive sessions, but as sessions are run as separate threads, it should be possible to detect when a thread fails and send ExecResult right away.

However, what if the _whole DVC_ has become unresponsive? e.g. dvc processing logic crashed without closing channel. Maybe I am overthinking it, and this corner case could be ignored.

Also, we still need to know that the agent has registered the exec session and that execution has started (Currecntly there is no "success" response after ExecSession request). Therefore we still need to send some message after initial exec request to start the session was sent. Maybe something like `ExecSessionStartedRsp` message with session_id will be sufficient. We need this message to show the session state correctly on the client. (Starting -> Running -> Finished)

@awakecoding thoughts?

Reply:
I see! Sounds like a heartbeat message could be good then. At least if the DVC is crashed, we have a way to detect that.

Reply:
Removed session-specific heartbeat, added heartbeat for a whole channel; added separate ExecStarted message